S/SCIP/build_tarballs.jl

```diff
@@ -0,0 +1,65 @@
+# Note that this script can accept some limited command-line arguments, run
+# `julia build_tarballs.jl --help` to see a usage message.
+using BinaryBuilder, Pkg
+
+name = "SCIP"
+version = v"0.1.0"
+
+# Collection of sources required to complete build
+sources = [
+    ArchiveSource("https://scip.zib.de/download/release/scipoptsuite-7.0.1.tgz", "971962f2d896b0c8b8fa554c18afd2b5037092685735d9494a05dc16d56ad422")
+]
+
+# Bash recipe for building across all platforms
+script = raw"""
+cd $WORKSPACE/srcdir
+cd scipoptsuite-7.0.1/
+mkdir build
+cd build/
+cmake -DCMAKE_INSTALL_PREFIX=$prefix -DCMAKE_TOOLCHAIN_FILE=${CMAKE_TARGET_TOOLCHAIN} -DCMAKE_BUILD_TYPE=Release -DPAPILO=0 ..
+make
+make install
+"""
+
+# These are the platforms we will build for by default, unless further
+# platforms are passed in on the command line
+platforms = [
+    Linux(:i686, libc=:glibc),
+    Linux(:x86_64, libc=:glibc),
+    Linux(:aarch64, libc=:glibc),
+    Linux(:armv7l, libc=:glibc, call_abi=:eabihf),
+    Linux(:powerpc64le, libc=:glibc),
+    Linux(:i686, libc=:musl),
+    Linux(:x86_64, libc=:musl),
+    Linux(:aarch64, libc=:musl),
+    Linux(:armv7l, libc=:musl, call_abi=:eabihf),
+    MacOS(:x86_64),
+    FreeBSD(:x86_64),
+    Windows(:i686)
+]
+
+platforms = expand_cxxstring_abis(platforms)
+
+# The products that we will ensure are always built
+products = [
+    ExecutableProduct("zimpl", :zimpl),
+    ExecutableProduct("scip", :scip),
+    ExecutableProduct("soplex", :soplex),
+    LibraryProduct("libgcg", :libgcg),
+    LibraryProduct("libscip", :libscip),
+    ExecutableProduct("gcg", :gcg)
+]
+
+# Dependencies that must be installed before this package can be built
+dependencies = [
+    Dependency(PackageSpec(name="bliss_jll", uuid="508c9074-7a14-5c94-9582-3d4bc1871065"))
+    Dependency(PackageSpec(name="GMP_jll", uuid="781609d7-10c4-51f6-84f2-b8444358ff6d"))
+    Dependency(PackageSpec(name="Ipopt_jll", uuid="9cc047cb-c261-5740-88fc-0cf96f7bdcc7"))
+    Dependency(PackageSpec(name="boost_jll", uuid="28df3c45-c428-5900-9ff8-a3135698ca75"))
```

Comment:
I believe boost is used as header only
```suggestion
    BuildDependency(PackageSpec(name="boost_jll", uuid="28df3c45-c428-5900-9ff8-a3135698ca75"))
```
I don't see any of the boost libraries in
```
% ldd bin/*
bin/gcg:
        linux-vdso.so.1 (0x00007ffc25bfd000)
        libscip.so.7.0 => /tmp/mozilla_mose0/x86_64-linux-gnu-cxx11/bin/../lib/libscip.so.7.0 (0x00007ff139c16000)
        libbliss.so => not found
        libgmp.so.10 => /usr/lib/libgmp.so.10 (0x00007ff139b30000)
        libgmpxx.so.4 => /usr/lib/libgmpxx.so.4 (0x00007ff139b27000)
        libstdc++.so.6 => /usr/lib/libstdc++.so.6 (0x00007ff13994a000)
        libm.so.6 => /usr/lib/libm.so.6 (0x00007ff139805000)
        libgcc_s.so.1 => /usr/lib/libgcc_s.so.1 (0x00007ff1397e9000)
        libc.so.6 => /usr/lib/libc.so.6 (0x00007ff139622000)
        libz.so.1 => /usr/lib/libz.so.1 (0x00007ff139608000)
        libreadline.so.8 => /usr/lib/libreadline.so.8 (0x00007ff1395b6000)
        libipopt.so.3 => not found
        libbliss.so => not found
        libquadmath.so.0 => /usr/lib/libquadmath.so.0 (0x00007ff139569000)
        /lib64/ld-linux-x86-64.so.2 => /usr/lib64/ld-linux-x86-64.so.2 (0x00007ff13a898000)
        libncursesw.so.6 => /usr/lib/libncursesw.so.6 (0x00007ff1394f8000)
bin/scip:
        linux-vdso.so.1 (0x00007ffcf59e0000)
        libz.so.1 => /usr/lib/libz.so.1 (0x00007faf50c77000)
        libreadline.so.8 => /usr/lib/libreadline.so.8 (0x00007faf50c25000)
        libgmp.so.10 => /usr/lib/libgmp.so.10 (0x00007faf50b84000)
        libgmpxx.so.4 => /usr/lib/libgmpxx.so.4 (0x00007faf50b7b000)
        libbliss.so => not found
        libipopt.so.3 => not found
        libm.so.6 => /usr/lib/libm.so.6 (0x00007faf50a34000)
        libquadmath.so.0 => /usr/lib/libquadmath.so.0 (0x00007faf509e9000)
        libstdc++.so.6 => /usr/lib/libstdc++.so.6 (0x00007faf5080c000)
        libgcc_s.so.1 => /usr/lib/libgcc_s.so.1 (0x00007faf507f2000)
        libc.so.6 => /usr/lib/libc.so.6 (0x00007faf5062b000)
        libncursesw.so.6 => /usr/lib/libncursesw.so.6 (0x00007faf505ba000)
        /lib64/ld-linux-x86-64.so.2 => /usr/lib64/ld-linux-x86-64.so.2 (0x00007faf50cd8000)
bin/soplex:
        linux-vdso.so.1 (0x00007ffc887ea000)
        libboost_program_options.so.1.71.0 => not found
        libz.so.1 => /usr/lib/libz.so.1 (0x00007f879f64b000)
        libgmp.so.10 => /usr/lib/libgmp.so.10 (0x00007f879f5aa000)
        libgmpxx.so.4 => /usr/lib/libgmpxx.so.4 (0x00007f879f5a1000)
        libquadmath.so.0 => /usr/lib/libquadmath.so.0 (0x00007f879f556000)
        libstdc++.so.6 => /usr/lib/libstdc++.so.6 (0x00007f879f379000)
        libm.so.6 => /usr/lib/libm.so.6 (0x00007f879f232000)
        libgcc_s.so.1 => /usr/lib/libgcc_s.so.1 (0x00007f879f218000)
        libc.so.6 => /usr/lib/libc.so.6 (0x00007f879f051000)
        /lib64/ld-linux-x86-64.so.2 => /usr/lib64/ld-linux-x86-64.so.2 (0x00007f879f6ac000)
bin/zimpl:
        linux-vdso.so.1 (0x00007ffe54333000)
        libgmp.so.10 => /usr/lib/libgmp.so.10 (0x00007fc09e5d9000)
        libgmpxx.so.4 => /usr/lib/libgmpxx.so.4 (0x00007fc09e5d0000)
        libz.so.1 => /usr/lib/libz.so.1 (0x00007fc09e5b6000)
        libm.so.6 => /usr/lib/libm.so.6 (0x00007fc09e471000)
        libc.so.6 => /usr/lib/libc.so.6 (0x00007fc09e2aa000)
        libstdc++.so.6 => /usr/lib/libstdc++.so.6 (0x00007fc09e0cd000)
        libgcc_s.so.1 => /usr/lib/libgcc_s.so.1 (0x00007fc09e0b1000)
        /lib64/ld-linux-x86-64.so.2 => /usr/lib64/ld-linux-x86-64.so.2 (0x00007fc09e6c1000)
% ldd lib/*.so
lib/libgcg.so:
        linux-vdso.so.1 (0x00007ffc717fd000)
        libscip.so.7.0 => /tmp/mozilla_mose0/x86_64-linux-gnu-cxx11/lib/libscip.so.7.0 (0x00007f34c5033000)
        libbliss.so => not found
        libgmp.so.10 => /usr/lib/libgmp.so.10 (0x00007f34c4f4d000)
        libgmpxx.so.4 => /usr/lib/libgmpxx.so.4 (0x00007f34c4f44000)
        libstdc++.so.6 => /usr/lib/libstdc++.so.6 (0x00007f34c4d67000)
        libm.so.6 => /usr/lib/libm.so.6 (0x00007f34c4c22000)
        libgcc_s.so.1 => /usr/lib/libgcc_s.so.1 (0x00007f34c4c06000)
        libc.so.6 => /usr/lib/libc.so.6 (0x00007f34c4a3f000)
        libz.so.1 => /usr/lib/libz.so.1 (0x00007f34c4a25000)
        libreadline.so.8 => /usr/lib/libreadline.so.8 (0x00007f34c49d3000)
        libipopt.so.3 => not found
        libbliss.so => not found
        libquadmath.so.0 => /usr/lib/libquadmath.so.0 (0x00007f34c4986000)
        /usr/lib64/ld-linux-x86-64.so.2 (0x00007f34c6158000)
        libncursesw.so.6 => /usr/lib/libncursesw.so.6 (0x00007f34c4915000)
lib/libscip.so:
        linux-vdso.so.1 (0x00007ffcf33b0000)
        libz.so.1 => /usr/lib/libz.so.1 (0x00007f61208d8000)
        libreadline.so.8 => /usr/lib/libreadline.so.8 (0x00007f6120886000)
        libgmp.so.10 => /usr/lib/libgmp.so.10 (0x00007f61207e5000)
        libgmpxx.so.4 => /usr/lib/libgmpxx.so.4 (0x00007f61207dc000)
        libipopt.so.3 => not found
        libbliss.so => not found
        libm.so.6 => /usr/lib/libm.so.6 (0x00007f6120695000)
        libquadmath.so.0 => /usr/lib/libquadmath.so.0 (0x00007f612064a000)
        libstdc++.so.6 => /usr/lib/libstdc++.so.6 (0x00007f612046d000)
        libgcc_s.so.1 => /usr/lib/libgcc_s.so.1 (0x00007f6120453000)
        libc.so.6 => /usr/lib/libc.so.6 (0x00007f612028c000)
        /usr/lib64/ld-linux-x86-64.so.2 (0x00007f61215b9000)
        libncursesw.so.6 => /usr/lib/libncursesw.so.6 (0x00007f612021b000)
```

Reply:
What about `libboost_program_options.so.1.71.0 => not found` below `bin/soplex`?
Not that `soplex` as an executable is really needed for SCIP.jl

Reply:
from the build log:
```
2020-08-04T07:15:40.1169369Z [0m[1m[07:15:40] [22m-- The following OPTIONAL packages have been found:
2020-08-04T07:15:40.1170706Z [0m[1m[07:15:40] [22m
2020-08-04T07:15:40.1171610Z [0m[1m[07:15:40] [22m * Quadmath
2020-08-04T07:15:40.1172743Z [0m[1m[07:15:40] [22m * Boost (required version >= 1.58.0)
2020-08-04T07:15:40.1173792Z [0m[1m[07:15:40] [22m * Readline
2020-08-04T07:15:40.1174701Z [0m[1m[07:15:40] [22m * ZLIB
2020-08-04T07:15:40.1175586Z [0m[1m[07:15:40] [22m * PkgConfig
2020-08-04T07:15:40.1176504Z [0m[1m[07:15:40] [22m * GMP
2020-08-04T07:15:40.1177378Z [0m[1m[07:15:40] [22m * BLISS
```
So it seems boost is needed as a runtime lib?